src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -1195,6 +1216,29 @@ public static TestEtlScriptResult TestScript<TC, TCS>(
                                                     "while ETL test expects to get exactly 1 transformation script");
             }
 
+            List<string> debugOutput;
+            if (testScript.Configuration.EtlType == EtlType.GenAi)
+            {
+                var testGenAiScript = testScript as TestGenAiScript;
+                var aiGenConfiguration = testGenAiScript?.Configuration;
+                using (var genAiTask = new GenAiTask(testScript.Configuration.Transforms[0], aiGenConfiguration, database, database.ServerStore))
+                using (genAiTask.EnterTestMode(out debugOutput))
+                {
+                    genAiTask.EnsureThreadAllocationStats();
+                    var result = genAiTask.RunTest(testGenAiScript, context);
+                    result.DebugOutput = debugOutput;
+                    return result;
+                }
+            }
+
+            using var tx = testScript.IsDelete
+                ? context.OpenWriteTransaction()
```

Comment:
I'm a bit worried about this, because this will hold the write tx lock

Reply:
Note that this is how we do that across all the ETLs that modify docs. 
This is only for testing purposes. not expected to be done on high load systems.